Userland/Libraries/LibWeb/Layout/TreeBuilder.cpp

```diff
@@ -339,6 +361,40 @@ ErrorOr<void> TreeBuilder::create_layout_tree(DOM::Node& dom_node, TreeBuilder::
         }
     }
 
+    // https://html.spec.whatwg.org/multipage/rendering.html#button-layout
+    // If the element is an input element, or if it is a button element and its computed value for
+    // 'display' is not 'inline-grid', 'grid', 'inline-flex', or 'flex', then the element's box has
+    // a child anonymous button content box with the following behaviors:
+    if (is<HTML::HTMLButtonElement>(dom_node) && !(display == CSS::Display::from_short(CSS::Display::Short::InlineGrid) || display == CSS::Display::from_short(CSS::Display::Short::Grid) || display == CSS::Display::from_short(CSS::Display::Short::InlineFlex) || display == CSS::Display::from_short(CSS::Display::Short::Flex))) {
```

Comment:
We can check `!display.is_grid_inside() && !display.is_flex_inside()` instead of comparing against the four shorthands :)